src/components/Map.js

```diff
@@ -1,20 +1,32 @@
 import React, { useContext, useState } from 'react';
 
 import GoogleMapReact from 'google-map-react';
+import { Dialog } from '@reach/dialog';
+import '@reach/dialog/styles.css';
 
 import { SearchResultsContext } from '../context/SearchResults';
 
 import Pin from './Pin';
+import DialogBody from './DialogBody';
 
 const Map = () => {
   const [loaded, setLoaded] = useState(false);
+  const [locationDetails, setLocationDetails] = useState(null);
   const { data } = useContext(SearchResultsContext);
   const { centerCoords, zoom } = data;
 
   const handleApiLoaded = () => {
     setLoaded(true);
   };
 
+  const handleClick = (location) => {
+    setLocationDetails(location);
+  };
+
+  const handleClose = () => {
+    setLocationDetails(null);
```

Comment:
I believe that I could understand 😄 what you want to do with `locationDetails` state (open/close a dialog). Then I have a little suggestion **(no pressure to do it 😄).**

Context: It's a difficult to understand (for me) when you set `null` to close a dialog. I would expect a boolean `true` / `false`. Like `setIsOpenDialog(true)` or `setIsOpenDialog(false)`.

Also, I think you can do: `{ isOpenDialog && <Dialog>....</Dialog> }`. 😄

Reply:
hey @alxmcr, the logic behind it is this:  on pin click, we set location details into state, we update them to null on closing the dialog. So I think we are good (as per your suggestion, it is another way of going about this but following it would add an extra piece of state and 2 extra calls to dialog state updater functions)